src/backends/backend/python/CMakeLists.txt

```diff
@@ -0,0 +1,134 @@
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+cmake_minimum_required (VERSION 3.5)
+
+#
+# libpython_interpreter.so
+#
+configure_file(libtriton_python.ldscript libtriton_python.ldscript COPYONLY)
+
+find_package(gRPC CONFIG REQUIRED)
+find_package(Protobuf CONFIG REQUIRED)
+
+set(gRPC_CPP_PLUGIN_EXECUTABLE $<TARGET_FILE:gRPC::grpc_cpp_plugin>)
+
+file(COPY ${CMAKE_CURRENT_SOURCE_DIR}/python_host.proto DESTINATION ${CMAKE_CURRENT_BINARY_DIR})
+set(Protobuf_SRC_ROOT_FOLDER ${CMAKE_CURRENT_BINARY_DIR})
+
+protobuf_generate_cpp(PROTO_SRCS PROTO_HOST_HDRS ${CMAKE_CURRENT_BINARY_DIR}/python_host.proto)
+
+protobuf_generate_python(PROTO_PY ${CMAKE_CURRENT_BINARY_DIR}/python_host.proto)
+
+set (protocol_grpc_srcs "${CMAKE_CURRENT_BINARY_DIR}/python_host.grpc.pb.cc")
+set (protocol_grpc_hdrs "${CMAKE_CURRENT_BINARY_DIR}/python_host.grpc.pb.h")
+set (protocol_grpc_py "${CMAKE_CURRENT_BINARY_DIR}/python_host_pb2_grpc.py")
+
+add_custom_command(
+      OUTPUT "${protocol_grpc_srcs}" "${protocol_grpc_hdrs}"
+      COMMAND protobuf::protoc
+      ARGS --grpc_out "${CMAKE_CURRENT_BINARY_DIR}"
+        --cpp_out "${CMAKE_CURRENT_BINARY_DIR}"
+        --plugin=protoc-gen-grpc="${gRPC_CPP_PLUGIN_EXECUTABLE}"
+        -I "${CMAKE_CURRENT_BINARY_DIR}"
+        "python_host.proto"
+        DEPENDS "${PROTO_SRCS}")
+
+add_custom_command(
+        OUTPUT "${protocol_grpc_py}"
+        COMMAND python3
+        ARGS -m grpc_tools.protoc
+        --python_out "${CMAKE_CURRENT_BINARY_DIR}"
+        --grpc_python_out "${CMAKE_CURRENT_BINARY_DIR}"
+        -I "${CMAKE_CURRENT_BINARY_DIR}"
+        "python_host.proto"
+        DEPENDS "${PROTO_SRCS}")
+
+add_custom_target(
+        python_libs ALL 
+        DEPENDS ${PROTO_PY} ${protocol_grpc_py})
+
+FILE(GLOB PROTOBUF_HEADERS ${CMAKE_CURRENT_BINARY_DIR}/*.cc)
+FILE(GLOB PROTOBUF_SRCS ${CMAKE_CURRENT_BINARY_DIR}/*.h)
+
+include_directories(${CMAKE_CURRENT_BINARY_DIR})
+
+add_library(
+  triton_python SHARED
+  python_interpreter.cc
+  ${PROTOBUF_HEADERS}
+  ${PROTOBUF_SRCS}
+  $<TARGET_OBJECTS:model-config-library>
+
+  ${protocol_grpc_hdrs}
+  ${protocol_grpc_srcs})
+
+if(${TRITON_ENABLE_GPU})
+  target_include_directories(triton_python PRIVATE ${CUDA_INCLUDE_DIRS})
+endif() # TRITON_ENABLE_GPU
+
+add_dependencies(triton_python model-config-library)
+
+set_target_properties(
+  triton_python
+  PROPERTIES LINK_DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/libtriton_python.ldscript
+)
+
+set_target_properties(
+  triton_python
+  PROPERTIES LINK_FLAGS "-Wl,--version-script libtriton_python.ldscript"
+)
+
+target_include_directories(
+        triton_python
+        PRIVATE ${CMAKE_CURRENT_BINARY_DIR})
+
+target_link_libraries(
+  triton_python
+  PRIVATE gRPC::grpc++
+)
+
+if(${TRTIS_ENABLE_GPU})
+target_include_directories(
```

Comment:
Isn't this already given above?

Reply:
It is not. The build failed in the CI complaining about `cuda_runtime_api` header.

Reply:
Yes, it was given. Sorry.